### ISSUE TITLE
Dictionary comprehension in swagger.py throws SyntaxError under python 2.6

### ISSUE THREAD

**Issue by jaraco** [yougov/pmxbot#28]
When attempting to run pmxbot under Python 2.6, pmxbot core commands fails to load because of wordnik/swagger.py:112. This is a line that's part of a dictionary comprehension only supported in python 2.7.

```
# pmxbot config.yaml 
Loading popular quotes
Loading dowski excuses
Loading pmxbot rolls
Loading pmxbot notifier
Loading pmxbot karma
Loading pmxbot config
Loading pmxbot system commands
Loading pmxbot quotes
Loading pmxbot core commands
Error initializing plugin pmxbot core commands = pmxbot.commands.
Traceback (most recent call last):
  File "/opt/pmxbot/lib/python2.6/site-packages/pmxbot/core.py", line 640, in _load_library_extensions
    init_func = ep.load()
  File "/opt/pmxbot/lib/python2.6/site-packages/setuptools-0.6c11-py2.6.egg/pkg_resources.py", line 1954, in load
    entry = __import__(self.module_name, globals(),globals(), ['__name__'])
  File "/opt/pmxbot/lib/python2.6/site-packages/pmxbot/commands.py", line 21, in <module>
    from . import util
  File "/opt/pmxbot/lib/python2.6/site-packages/pmxbot/util.py", line 15, in <module>
    import wordnik.swagger
  File "/opt/pmxbot/lib/python2.6/site-packages/wordnik/swagger.py", line 112
     for (key, val) in objDict.iteritems()
       ^
 SyntaxError: invalid syntax
Loading pmxbot feedparser
....
```

----------------------------------------
- Bitbucket: https://bitbucket.org/yougov/pmxbot/issue/28
- Originally reported by: Anonymous
- Originally created at: 2013-06-27T05:44:21.637

**Comment by jaraco** [comment]
Thanks for the report. pmxbot dropped support for Python 2.6 in c3cf374fbedd. Unfortunately, I don't control wordnik. If you can file the ticket with the [upstream project](https://github.com/wordnik/wordnik-python) such that they agree to support Python 2.6, I'd be inclined to continue to support 2.6. Until then, pmxbot will only support Python 2.7+.

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
On second thought, there's already support for trapping ImportErrors - it's probably reasonable to trap the SyntaxError as well.

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Trap SyntaxError in wordnik.swagger (2.1.2). Fixes #28.

→ <<cset a2b9da7cec47>>

----------------------------------------
Original comment by: Jason R. Coombs
